File: fvc-api/src/main/java/sep490g65/fvcapi/service/FieldService.java
package sep490g65.fvcapi.service;

import org.springframework.transaction.annotation.Transactional;
import sep490g65.fvcapi.dto.request.CreateFieldRequest;
import sep490g65.fvcapi.dto.request.RequestParam;
import sep490g65.fvcapi.dto.request.UpdateFieldRequest;
import sep490g65.fvcapi.dto.response.FieldResponse;
import sep490g65.fvcapi.dto.response.PaginationResponse;

public interface FieldService {

    @Transactional(readOnly = true)
    PaginationResponse<FieldResponse> list(RequestParam params);

    FieldResponse getById(String id);
    
    FieldResponse create(CreateFieldRequest request);
    
    FieldResponse update(String id, UpdateFieldRequest request);
    
    void delete(String id);
}

File: fvc-frontend/src/services/fieldService.ts
import apiService from './api';
import { API_ENDPOINTS } from '../config/endpoints';
import type { BaseResponse, PaginationResponse } from '../types/api';
import type { CreateFieldRequest, UpdateFieldRequest, FieldFilters, FieldResponse } from '../types';

export const fieldService = {
  async list(params: FieldFilters): Promise<PaginationResponse<FieldResponse>> {
    const res = await apiService.get<PaginationResponse<FieldResponse>>(API_ENDPOINTS.FIELDS.BASE, params);
    
    if (!res.data) {
      console.error('FieldService - response.data is undefined!');
      throw new Error('Invalid API response structure');
    }
    return res.data;
  },

  async get(id: string): Promise<FieldResponse> {
    const res = await apiService.get<BaseResponse<FieldResponse>>(API_ENDPOINTS.FIELDS.BY_ID(id));
    return res.data.data;
  },

  async create(payload: CreateFieldRequest): Promise<FieldResponse> {
    const res = await apiService.post<BaseResponse<FieldResponse>>(API_ENDPOINTS.FIELDS.BASE, payload);
    return res.data.data;
  },

  async update(id: string, payload: UpdateFieldRequest): Promise<FieldResponse> {
    const res = await apiService.put<BaseResponse<FieldResponse>>(API_ENDPOINTS.FIELDS.BY_ID(id), payload);
    return res.data.data;
  },

  async remove(id: string): Promise<void> {
    await apiService.delete<void>(API_ENDPOINTS.FIELDS.BY_ID(id));
  },
};

